src/Data/Functor/Day/Curried.hs

```diff
@@ -37,6 +37,9 @@ import Control.Applicative
 import Data.Functor.Adjunction
 import Data.Functor.Day
 import Data.Functor.Identity
+#if MIN_VERSION_base(4,10,0)
```

Comment:
Since we already guard an import of `Control.Applicative` with CPP above, I'd prefer to combine the CPP into one place. Something like this:

```hs
#if !(MIN_VERSION_base(4,8,0)) || MIN_VERSION_base(4,10,0)
import Control.Applicative
#endif
```

Here, I'm replacing the use of `#if __GLASGOW_HASKELL__ < 710` above with `#if !(MIN_VERSION_base(4,8,0))`, which is slightly more accurate.

Reply:
@RyanGlScott, let's just `import Control.Applicative (Applicative (..))` unconditionally, and not worry about the unused import warnings in the in-between versions.

Reply:
I'd prefer to keep the build warning-free. I'm fine with merging it as-is, but I'd just add the CPP back afterwards anyway.